tabular/src/autogluon/tabular/models/tabular_nn/torch/tabular_torch_dataset.py

```diff
@@ -122,17 +122,31 @@ def __init__(self, processed_array, feature_arraycol_map, feature_type_map, prob
         self.num_categories_per_embed_feature = None
         self.num_categories_per_embedfeature = self.getNumCategoriesEmbeddings()
 
-    def __getitem__(self, idx):
-        output_dict = {}
-        if self.has_vector_features():
-            output_dict['vector'] = self.data_list[self.vectordata_index][idx]
-        if self.num_embed_features() > 0:
-            output_dict['embed'] = []
-            for i in self.embed_indices:
-                output_dict['embed'].append(self.data_list[i][idx])
-        if self.label_index is not None:
-            output_dict['label'] = self.data_list[self.label_index][idx]
-        return output_dict
+        self.has_vector_features = self.vectordata_index is not None
+        self.has_embed_features = len(self.feature_groups['embed']) > 0
+
+    def __iter__(self):
```

Comment:
Add some documentation here since the code is a bit complex

Reply:
Documentation added.